docs/docs/debugging-async-lifecycles.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Debugging Asynchronous Lifecycle Methods
+---
+
+Various lifecycle methods (see: [Gatsby Node APIs](/docs/node-apis/)) within Gatsby are presumed to be asynchronous. In other words, these methods can _eventually_ resolve to a value and this value is a [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). We wait for the `Promise` to resolve, and then mark the lifecycle method as completed when it does.
```

Comment:
This can be misleading - we don't presume that APIs are async, but we allow them to be - and if they are they need to return Promise.

`createPages` can be sync:
```js
exports.createPages = ({ actions }) => {
  const data = require('./some-data.json')
  data.forEach(datum => actions.createPage({ path: datum.path, component: <some_component_path> }))
}
```
is correct and will work just fine.

Reply:
I don't think this is misleading necessarily. They _are_ presumed to be async.

Reply:
How should I tweak the language though?

Reply:
I think we should drop my comment to ship this and iterate if needed.